### ISSUE TITLE
Chrx fails on ChromeOS 85

### ISSUE THREAD

**Issue by jonasthenoob** [reynhout/chrx#98]
im having this problem when i run the command to install it
and its weird cause i have ctrl-L seabios booting enabled[
![Screenshot 2020-07-06 at 21 24 22](https://user-images.githubusercontent.com/57160032/86642702-04a33c00-bfd4-11ea-8586-1956111653da.png)

**Comment by reynhout** [comment]
Interesting. Thanks for the info!

Not sure what's going on yet, I'll see if I can reproduce.

That message is displayed when the check for RW_LEGACY firmware fails. The command to check is `/usr/sbin/mosys eeprom map | grep RW_LEGACY`. This has worked for many years, and has definitely worked on EDGAR...it's possible that a new ChromeOS update alters behavior...? I'll see what I can find.

**Comment by jonasthenoob** [comment]
No worries @reynhout, a quick powerwash fixed this issue im not sure why it happened to begin with but thanks for looking into it really means a lot.

**Comment by reynhout** [comment]
@jonasthenoob Thanks for the update, and glad you got it working! FWIW, I don't have an EDGAR, but I was not able to repro on a different Braswell machine...sounds like your device was just in a wonky state.

**Comment by justathoughtor2** [comment]
I'm actually having the same exact error message show up on Pixelbook (eve). This wasn't an issue when I last ran the chrx script in April, so either the May update to 3.0.1 broke the script for me, or a Chrome OS update broke it. I'm on Chrome OS 85 (beta). I'll try downgrading to stable and see if I can get chrx to work. I'll report back with my findings.

**Comment by reynhout** [comment]
@justathoughtor2 Interesting. I don't think there are any relevant changes in chrx, so I lean toward a ChromeOS change. But please let us know what you learn!

**Comment by reynhout** [comment]
The ChromeOS command that chrx wants to use is `/usr/sbin/mosys eeprom map | grep RW_LEGACY`.

If this fails, chrx will report the given error message. There are a few reasons it might fail -- e.g. the `mosys` command might be moved (or removed), or a new version might require different parameters. If you hit the error again, and are able to pastebin the output of `sudo /usr/bin/mosys`, that would be a great help in debugging!

**Comment by justathoughtor2** [comment]
@reynhout Sorry, didnt get the output of that command before downgrading to stable. But downgrading to stable resolved the issue; I was able to successfully run chrx as normal. Definitely seems to be an incompatibility with Chrome OS 85 (and possibly higher).

**Comment by reynhout** [comment]
@justathoughtor2 Got it, that's great info, thank you!

I'll reopen this ticket and test on the newest ChromeOS.

**Comment by RedcatPhoenix2** [comment]
I've figured out what caused the problem. It only supports 84 and under right now.

**Comment by RedcatPhoenix2** [comment]
So... It works on Chrome OS 84 Stable and under (Chrome OS 84 beta/dev also)

**Comment by iceblink** [comment]
That mosys command was removed here:
https://chromium-review.googlesource.com/c/chromiumos/platform/mosys/+/2216234

You can accomplish something similar with the dump_fmap command:
```
flashrom -r /tmp/bios.bin
dump_fmap -p /tmp/bios.bin RW_LEGACY
```

**Comment by reynhout** [comment]
@iceblink That makes perfect sense. Thank you!

**Comment by reynhout** [comment]
This should be fixed in chrx-3.0.2. Thanks for the help!
